posts/what-makes-entando-a-highly-avialable-architecture.md

```diff
@@ -0,0 +1,81 @@
+
+Caching techniques can also help scale database reads. Storing frequently accessed data in a separate cache keeps the load off of the database.  When a user or an application goes to access the data, it may not even hit the database, pulling from the cache instead.
+
+A third way to scale database reads is to implement database sharding. This strategy is also the primary method used to scale database writes.
+
+### Scaling Database Writes
+
+A database write happens when a user or program wishes to store a piece of data on the database. Since every database replica will also need to contain this piece of data, increasing replicas does not fix the issue as it does for database reads. Enter what is known as "database sharding."
+
+Sharding a database is the process of spreading the data across multiple databases. Spreading the data out can be as simple as putting records that start with the character 'A' through 'L' in one database and the remaining 'M' to 'Z' records in a different database. This arrangement means we need another component in front of the database to handle the routing. Some databases support sharding natively, and others will require a third-party tool.
+
+### Scaling Database Automation in Kubernetes
+
+Scaling a database in Kubernetes is similar to how we would for a cluster. Adding additional cluster nodes that run the database is the best way. And it is crucial to complete this scaling task before the added traffic makes its way to the new cluster(s). The process includes the following:
+- Creating a new node.
+- Initiating a backup save of the database.
+- Initiating a backup restore on the new node.
+
+# Many Masters
+
+All this talk about scaling nodes and refreshing pods begs the question, what manages these resources? The answer is the Kubernetes "Master Node," also known as the "Primary Node." Scaling the worker nodes provides Kubernetes with more resources. More worker nodes bring more CPU and memory resources for pod scheduling. What about scaling the Primary Node itself? There are many benefits to scaling the Primary Node in addition to the worker nodes.
+
+Configuring Kubernetes to have multiple Primary Nodes will bulletproof your application deployment. As mentioned before, Entando allows you to do this and supports it with minimal configuration. The recommended minimum number of Primary Nodes for a highly available Kubernetes deployment is 3.
+
+Having multiple masters also improves the network performance of the cluster because the multiple masters act together as a unified data center.  This all serves to reduce the possible failure of the etcd datastore, API server, controller manager and scheduler, and all of the worker nodes.  In a single Master Node setup
```

Comment:
I think you did not finish your last sentence :)

Reply:
Whoops!